File: ems_interface/src/main/java/com/ems/service/DocumentService.java
package com.ems.service;
import java.util.List;

import com.ems.pojo.Document;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface DocumentService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Document> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Document document);
	
	
	/**
	 * 修改
	 */
	public void update(Document document);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Document findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Document document, int pageNum, int pageSize);
	
}

File: ems_user_web/src/main/webapp/js/service/documentService.js
//服务层
app.service('documentService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../document/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../document/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../document/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../document/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../document/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../document/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../document/search.do?page='+page+"&rows="+rows, searchEntity);
	}


	//查询天气
	this.cha=function (city) {
		return $http({
			method:'jsonp',
			url:'http://api.map.baidu.com/telematics/v3/weather',
			params: {
				callback:"JSON_CALLBACK",
                location: city,
                output: 'json',
                ak: '0A5bc3c4fb543c8f9bc54b77bc155724'
			}
		});
    }
});
